src/tlo/analysis/life_expectancy.py

```diff
@@ -0,0 +1,305 @@
+"""
+This file will read in the output files generated by model runs
+then employs a series of helper functions to
+generate life tables to estimate life expectancy for each run/draw
+
+function: produce_life_expectancy_estimates() will produce summary statistics
+or will output life expectancy at birth for M and F separately for every
+run/draw if argument(...median=False)
+
+"""
+
+from pathlib import Path
+
+import pandas as pd
+
+from tlo import Date
+from tlo.analysis.utils import extract_results, get_scenario_info, load_pickled_dataframes
+
+# define the target period over which we want to estimate life expectancy
+# this should be a one-year period
+TARGET_PERIOD = (Date(2019, 1, 1), Date(2020, 1, 1))
+
+
+# %% --------------------------------------------------------------
+# HELPER FUNCTIONS
+
+
+def _map_age_to_age_group(dataframe):
+    """
+    Maps ages to age-groups in 5-year intervals and adds a new column 'age-group' to the DataFrame.
+
+    Args:
+    - dataframe (pd.DataFrame): The DataFrame containing the age data.
+    - age_column (str): The name of the column containing ages.
+
+    Returns:
+    - pd.DataFrame: The DataFrame with the 'age-group' column added.
+    """
+    # Define age groups in 5-year intervals
+    age_groups = ['0'] + ['1-4'] + [f'{start}-{start + 4}' for start in range(5, 90, 5)] + ['90+']
+
+    # Create a new column 'age-group' based on the age-to-age-group mapping
+    dataframe['age_group'] = pd.cut(dataframe['age'], bins=[0] + [1] + list(range(5, 95, 5)) + [float('inf')],
+                                    labels=age_groups, right=False)
+
+    return dataframe
```

Comment:
I think more natural if this function returns a `pd.Series` contains the mapped age-groups, rather than directly editing the data frame in place.

Reply:
This is changed now and the new dataframe column is set within the function where it is needed